src/modules/tray.cpp

```diff
@@ -0,0 +1,50 @@
+//
+// Created by raffael on 02.02.22.
+//
+
+#include "modules/tray.hpp"
+
+#include "modules/meta/base.inl"
+
+POLYBAR_NS
+namespace modules {
+  template class module<tray_module>;
+
+  tray_module::tray_module(const bar_settings& bar_settings, string name_)
+      : event_module<tray_module>(bar_settings, move(name_)) {
+    m_formatter->add(DEFAULT_FORMAT, "", {});
```

Comment:
To be consistent, we should also use `format = <tray>` for the tray. This here just adds a pseudo format that doesn't really do anything (since it has no tags).

```suggestion
    m_formatter->add(DEFAULT_FORMAT, TAG_TRAY, {TAG_TRAY});
```

And `TAG_TRAY` could be defined as `<tray>`.

In that case, you also don't need `get_output` and can just check for `TAG_TRAY` in `tray_module::build`. See how the script module does this: https://github.com/polybar/polybar/blob/ab915fb724546e7abfcb35cbaf863ce2bca613ec/src/modules/script.cpp#L114-L122

Reply:
You can completely remove `get_output`. There is already a `get_output` defined for all modules in `base.inl` which takes care of invoking the `build` function.